kernel/Kconfig

```diff
@@ -937,7 +937,6 @@ config TICKLESS_KERNEL
 
 config TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE
 	bool
-	default y if "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr" || "$(ZEPHYR_TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE)" = "y"
```

Comment:
a bit unfortunate that https://github.com/zephyrproject-rtos/zephyr/pull/45656 introduced `ZEPHYR_TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE` as environment variable, but now that's it's there we should do a proper deprecation and tell users how to define that the toolchain is supporting this.

I assume @keith-packard was using the `cross-compile` toolchain setting when testing with Debian arm-none-eabi, correct @keith-packard ?

The `"$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr" ` is something that should have been placed in the Zephyr SDK Kconfig file, not sure why I didn't noticed that back then: https://github.com/zephyrproject-rtos/zephyr/pull/29591
We should move this to Zephyr-SDK for the upcomming `0.16` release, and then make a comment here, that this code is kept for backwards compatibility until minimum required Zephyr SDK is 0.16.
fyi: @stephanosio

Reply:
It already is part of Zephyr SDK.

https://github.com/zephyrproject-rtos/sdk-ng/blob/0bc76f4b82e1f2d5d9d22f8a2c4fe9b1551cd08d/cmake/zephyr/Kconfig#L7

I believe this was a leftover from before the config was added on the SDK side.

Reply:
So only question remains is the `ZEPHYR_TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE` in environment.

Reply:
I put back `ZEPHYR_TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE` support.  We can clean that up in a separate PR based on what @keith-packard says.  (if we keep it we should document it since its a completely hidden setting).

Reply:
> a bit unfortunate that #45656 introduced `ZEPHYR_TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE` as environment variable, but now that's it's there we should do a proper deprecation and tell users how to define that the toolchain is supporting this.

Agreed -- I think having any environment variables adjusting configuration at this level is fraught with potential conflict.

> I assume @keith-packard was using the `cross-compile` toolchain setting when testing with Debian arm-none-eabi, correct @keith-packard ?

Yes, I regularly test Zephyr builds with the Debian toolchain as I am the most active maintainer for that, and Zephyr offers another great source of test cases, in addition to the glibc and picolibc test cases already in use.

Reply:
> Agreed -- I think having any environment variables adjusting configuration at this level is fraught with potential conflict.

@keith-packard just to be clear, you are good w/removing the env var option?

Reply:
As long as there's *some* way to configure this option, yes, removing the environment variable would be fine. I'm using Zephyr as part of the Debian toolchain validation process; changing the scripts there to adapt to whatever Zephyr needs is easy for me, it's not like that's part of a large number of product build systems.